IBX-9737: Collaborative editing described in Developer Documentation - v5.0 (#2721)
ezsystems/developer-documentation: docs/content_management/collaborative_editing/invitation_api.md

---
description: You can use the PHP API to create new invitation, update existing one, read or delete it.
---

# Invitation API

[`InvitationService`](../../api/php_api/php_api_reference/classes/Ibexa-Contracts-Core-Repository-InvitationService.html) enables you to read, add, update, and remove invitation for collaborative editing session.

## Create invitation

You can create new invitation for the collaborative session using the [`InvitationService::createInvitation`](../../api/php_api/php_api_reference/classes/Ibexa-Contracts-Core-Repository-InvitationService.html#method_createInvitation) method:

``` php

> {
>     $session = self::getSessionService()->getSession(self::EXAMPLE_SESSION_ID);
>     $participant = $session->getParticipants()->getById(self::EXAMPLE_PARTICIPANT_ID);
> 
>     $createStruct = new InvitationCreateStruct($session, $participant);
>     $createStruct->setContext([
>         'message' => 'Hello, would you like to join my session?',
>     ]);
> 
>     $invitation = $this->getInvitationService()->createInvitation($createStruct);
> 
>     self::assertEquals(InvitationStatus::STATUS_PENDING, $invitation->getStatus());
>     self::assertEquals([
>         'message' => 'Hello, would you like to join my session?',
>     ], $invitation->getContext());

Reviewer: ```suggestion
{
    $session = $this->sessionService->getSession(1);
    $participant = $session->getParticipants()->getByEmail('foo@link.invalid');

    $createStruct = new InvitationCreateStruct($session, $participant);
    $createStruct->setContext([
        'message' => 'Hello, would you like to join my session?',
    ]);

    $invitation = $this->invitationService->createInvitation($createStruct);
```

}
```

> ## Read invitation

Reviewer: ```suggestion
## Get invitation by id
```

> You can read an invitation with [`InvitationService::getInvitation`](../../api/php_api/php_api_reference/classes/Ibexa-Contracts-Core-Repository-InvitationService.html#method_deleteInvitation):

Reviewer: ```suggestion
You can get an invitation with [`InvitationService::getInvitation`](../../api/php_api/php_api_reference/classes/Ibexa-Contracts-Core-Repository-InvitationService.html#method_deleteInvitation):
```

``` php

>     $this->getInvitationService()->getInvitation(self::EXAMPLE_INVITATION_ID);

Reviewer: ```suggestion
    $this->invitationService->getInvitation(1);
```

```

You can select the parameter that you can read from an invitaion:

- Invitation ID:

``` php

>     self::assertEquals(self::EXAMPLE_INVITATION_ID, $invitation->getId());

Reviewer: ```suggestion
    $invitation->getId();
```

```

- Session ID:

``` php

>     self::assertEquals(self::EXAMPLE_SESSION_ID, $invitation->getSession()->getId());

Reviewer: ```suggestion
    $invitation->getSession()->getId();
```

```

- Participant ID:

``` php

>     self::assertEquals(self::EXAMPLE_PARTICIPANT_ID, $invitation->getParticipant()->getId());

Reviewer: ```suggestion
    $invitation->getParticipant()->getId();
```

```
    
- Invitation status:

``` php

>     self::assertEquals(InvitationStatus::STATUS_PENDING, $invitation->getStatus());

Reviewer: ```suggestion
    $invitation->getStatus();
```

```

> ## Find invitation

Reviewer: ```suggestion
## Find invitations
```

You can find an invitation with [`InvitationService::findInvitation`](../../api/php_api/php_api_reference/classes/Ibexa-Contracts-Core-Repository-InvitationService.html#method_deleteInvitation) by:

- Invitation ID:

``` php
[[= include_file('code_samples/api/public_php_api/src/Command/(?).php', , ) =]]
```

- Session ID:

- Participant ID:

> - Invitation status:

Reviewer: Here are Criteria which allows to find invitations: https://github.com/ibexa/collaboration/tree/4.6/src/contracts/Invitation/Query/Criterion

## Update invitation

You can update existing invitation with [`InvitationService::deleteInvitation`](../../api/php_api/php_api_reference/classes/Ibexa-Contracts-Core-Repository-InvitationService.html#method_deleteInvitation):

``` php

> $invitation = $this->getInvitationService()->getInvitation(self::EXAMPLE_INVITATION_ID);

Reviewer: ```suggestion
$invitation = $this->invitationService->getInvitation(1);

$updateStruct = new InvitationUpdateStruct();
$updateStruct->setStatus(InvitationStatus::STATUS_ACCEPTED);

$invitation = $this->invitationService->updateInvitation($invitation, $updateStruct);
```

```

## Delete invitation

You can delete an invitation with [`InvitationService::deleteInvitation`](../../api/php_api/php_api_reference/classes/Ibexa-Contracts-Core-Repository-InvitationService.html#method_deleteInvitation):

``` php

>     $this->getInvitationService()->deleteInvitation($invitation);

Reviewer: ```suggestion
$invitation = $this->invitationService->getInvitation(1);

$this->invitationService->deleteInvitation($invitation);
```

> ```

Reviewer: Method InvitationService::getInvitationByParticipant is not covered https://github.com/ibexa/collaboration/blob/4.6/src/contracts/InvitationServiceInterface.php
